proto/panacea/datapool/v2/genesis.proto

```diff
@@ -9,12 +9,11 @@ import "panacea/datapool/v2/pool.proto";
 
 // GenesisState defines the datapool module's genesis state.
 message GenesisState {
-  repeated panacea.datapool.v2.DataValidator data_validators = 1;
-  uint64                                     next_pool_number = 2;
-  repeated Pool                              pools = 3;
-  Params                                     params = 4 [
-    (gogoproto.nullable) = false
-  ];
+  repeated DataValidator data_validators = 1 [(gogoproto.nullable) = false];
+  uint64                 next_pool_number = 2;
+  repeated Pool          pools = 3 [(gogoproto.nullable) = false];
+  Params                 params = 4 [(gogoproto.nullable) = false];
+  repeated WhiteList     white_list = 5 [(gogoproto.nullable) = false];
```

Comment:
I also add `[ (gogoproto.nullable) = false ]` to some fields. 

When I export `genesis.json` with multiple values of those fields, all the same values (I guess the last one) are stored and exported. I referred to other projects and they all used the gogoproto extension. I'm not sure of the reason, it was a workaround for now. If you have any idea, please leave some comments.

Reply:
Are you saying this?
**expected:** 
```json
{
  "whiteList": [
    {
      "data_pool_code_id" : 1,
      "data_pool_nft_contract_adress": "panaceaxxxx1"
    },
    {
      "data_pool_code_id" : 2,
      "data_pool_nft_contract_adress": "panaceaxxxx2"
    }
  ]
}
```

**acutally**
```json
{
  "whiteList": [
    {
      "data_pool_code_id" : 2,
      "data_pool_nft_contract_adress": "panaceaxxxx2"
    }
  ]
}
```

Reply:
In your example,

```json
{
  "whiteList": [
    {
      "data_pool_code_id" : 2,
      "data_pool_nft_contract_adress": "panaceaxxxx2"
    },
    {
      "data_pool_code_id" : 2,
      "data_pool_nft_contract_adress": "panaceaxxxx2"
    }
  ]
}
```

The last one was duplicated..

Reply:
That's interesting...

Reply:
There are also differences of types in `GenesisState`. Without gogoproto extension, pointer is added to each type.

Without `[(gogoproto.nullable) = false]`, 
```go
type GenesisState struct {
    DataValidators []*DataValidator `protobuf:"bytes,1,rep,name=data_validators,json=dataValidators,proto3" json:"data_validators"`
    NextPoolNumber uint64           `protobuf:"varint,2,opt,name=next_pool_number,json=nextPoolNumber,proto3" json:"next_pool_number,omitempty"`
    Pools          []*Pool          `protobuf:"bytes,3,rep,name=pools,proto3" json:"pools"`
    Params         *Params          `protobuf:"bytes,4,opt,name=params,proto3" json:"params"`
}
```

With `[(gogoproto.nullable) = false]`, 
```go
type GenesisState struct {
    DataValidators []DataValidator `protobuf:"bytes,1,rep,name=data_validators,json=dataValidators,proto3" json:"data_validators"`
    NextPoolNumber uint64          `protobuf:"varint,2,opt,name=next_pool_number,json=nextPoolNumber,proto3" json:"next_pool_number,omitempty"`
    Pools          []Pool          `protobuf:"bytes,3,rep,name=pools,proto3" json:"pools"`
    Params         Params          `protobuf:"bytes,4,opt,name=params,proto3" json:"params"`
}
```

Reply:
I just read all changes again. Thank you.

Is this issue still being raised? Then, I will take a look at it.

Reply:
> There are also differences of types in GenesisState. Without gogoproto extension, pointer is added to each type.

This behavior is intended. But still, it's weird that the list contains all the same value.

Reply:
> Is this issue still being raised? Then, I will take a look at it.

This issue is resolved now, but I don't know what's going on.

> This behavior is intended.

Oh I see. I'm curious this part as well. I'll look at gogoproto extension more.

Reply:
I just removed `[(gogoproto.nullable) = false]` and checked that the exported list in the genesis contains two different elements. As I know, `[(gogoproto.nullable) = false]` shouldn't violate the fundamental behavior of protobuf marshalling/unmarhalling.

In protobuf, the nullability is a bit tricky to handle because everything is optional by default in protobuf 3. That's why all fields are defined with pointer such as `[]*DataValidator` or `*Params`. Previously, the protobuf 2 had the `required` keyword which sets some field not nullable, but that feature was deleted in protobuf 3: https://github.com/protocolbuffers/protobuf/issues/2497.

That's why the gogoproto, which is an extension of protobuf, introduced a new feature `nullable`. If you really want to make each list element not nullable, then you can use `[(gogoproto.nullable) = false]` so that gogoproto defines the list as `[]Element`, not `[]*Element`. I also agree with using it.

Reply:
> But still, it's weird that the list contains all the same value.

I guess some Go code wasn't implemented correctly at that time.

Reply:
Oh, I see. I'm very thankful to you for your detailed explanation.

>I guess some Go code wasn't implemented correctly at that time.

I think it could be, because as I remember It was exported correctly when I implement `ExportGenesis()` of module params in #294.